src/sd/spi.rs

```diff
@@ -1,32 +1,23 @@
 use crate::sys::*;
 
-use core::{marker::PhantomData, ops::Deref};
+use core::{borrow::BorrowMut, marker::PhantomData, ops::Deref};
 
 use esp_idf_hal::{
     gpio::{InputPin, OutputPin},
     peripheral::Peripheral,
-    spi::SpiDriver,
+    spi::{SpiDeviceDriver, SpiDriver},
 };
 
 pub struct SpiDevice<'d> {
     configuration: sdspi_device_config_t,
-    driver: SpiDriver<'d>,
+    host: spi_host_device_t,
     _p: PhantomData<&'d mut ()>,
 }
 
-impl Drop for SpiDevice<'_> {
-    fn drop(&mut self) {
-        let result = unsafe { spi_bus_free(self.get_host()) };
-
-        if result != ESP_OK {
-            panic!("Failed to remove SPI device");
-        }
-    }
-}
-
 impl<'d> SpiDevice<'d> {
-    pub fn new(
-        driver: SpiDriver<'d>,
+    pub fn new<T>(
+        host: spi_host_device_t,
```

Comment:
This is incorrect and regressing the existing code: by replacing `SpiDriver` with a raw `spi_host_device_t` you just lost any warranty, that `spi_host_device_t` will be valid until your `SpiDevice` is valid! By using `SpiDriver` or `Borrow<SpiDriver>` or `BorrowMut<SpiDriver>` you have the warranty, that the user cannot drop the `SpiDriver` while your own `SpiDevice` is still alive.